README.md

```diff
@@ -71,103 +104,267 @@ def custom_fallback(feature_name: str, context: dict) -> bool:
 
-Powershell:
+client = UnleashClient(
+    url="https://YOUR-API-URL",
+    app_name="my-python-app",
+    custom_headers={'Authorization': '<API token>'},
+    event_callback=example_callback
+)
+client.initialize_client()
+client.is_enabled("testFlag")
 
-```
-python -m venv venv
-venv\Scripts\activate.bat
 ```
 
-Once you've done your setup, run:
-```
-pip install -r requirements.txt
-```
+Impression callbacks run in-process — keep them fast to avoid blocking your app.
 
-Run the get-spec script to download the client specifications tests:
-```
-./scripts/get-spec.sh
-```
+### Custom cache
 
-Now you can run the tests by running `pytest` in the root directory.
+By default, the Python SDK stores feature toggles in an on-disk cache using fcache. If you need a different storage backend, for example, Redis, memory-only, or a custom database, you can provide your own cache implementation.
 
-In order to run a single test, run the following command:
+Below is an example custom CustomCache using fcache under the hood.
 
+```python
+from typing import Optional, Any
+from UnleashClient.cache import BaseCache
+from fcache.cache import FileCache as _Filecache
+
+class CustomCache(BaseCache):
+    # This is specific for FileCache.  Depending on the cache you're using, this may look different!
+    def __init__(self, name: str, directory: Optional[str] = None):
+        self._cache = _FileCache(name, app_cache_dir=directory)
+
+    def set(self, key: str, value: Any):
+        self._cache[key] = value
+        self._cache.sync()
+
+    def mset(self, data: dict):
+        self._cache.update(data)
+        self._cache.sync()
+
+    def get(self, key: str, default: Optional[Any] = None):
+        return self._cache.get(key, default)
+
+    def exists(self, key: str):
+        return key in self._cache
+
+    def destroy(self):
+        return self._cache.delete()
 ```
-pytest testfile.py::function_name
+Pass your cache instance to the client with the cache argument:
 
-# example: pytest tests/unit_tests/test_client.py::test_consistent_results
+```python
+client = UnleashClient(
+    url="https://YOUR-API-URL",
+    app_name="my-python-app",
+    cache=CustomCache("my-cache")
+)
 ```
 
-### Linting
+## Running in multi-process setups
 
-In order to lint all the files you can run the following command:
+The Python SDK runs a background thread to keep feature toggles in sync with the Unleash server. Some runtime environments, like WSGI servers and Celery workers, need extra setup to make sure the SDK works correctly.
 
+### WSGI
+
+When using WSGI servers (e.g., for Flask or Django apps), be aware that:
+
+- Many WSGI setups disable threading by default. The SDK needs threads to poll for updates in the background.
+- Make sure to set `enable-threads` in your WSGI config.
+
+If you use multiple processes (processes flag) for scaling, you may need to enable the `lazy-apps` flag. This ensures each process gets a fresh SDK instance.
```

Comment:
Would this work? Flags confuses me, even though it might be right word
```suggestion
When running under uWSGI with multiple processes (using the `--processes` option), you may need to enable the `lazy-apps` option. This ensures each process gets a fresh SDK instance.
```

Reply:
Yeah fair, I think that works nicely